pkg/provider/kubernetes/ingress/kubernetes.go

```diff
@@ -315,6 +315,12 @@ func (p *Provider) loadConfigurationFromIngresses(ctx context.Context, client Cl
 			}
 
 			for _, pa := range rule.HTTP.Paths {
+				if pa.Backend.Resource != nil {
+					// https://kubernetes.io/docs/concepts/services-networking/ingress/#resource-backend
+					logger.Error().Msg("Resource backends are not supported")
+					continue
+				}
+
 				service, err := p.loadService(client, ingress.Namespace, pa.Backend)
 				if err != nil {
 					logger.Error().
```

Comment:
Probably still shouldn't assume that pa.Backend.Service is non-nil here, even with the check for pa.Backend.Resource above.

Reply:
Thanks for pointing this out, we pushed a review commit for avoiding confusion and improve readability, but this case should never happen, as such an Ingress is invalid and this error should be raised: 
```The Ingress “ingress” is invalid: spec.rules[0].http.paths[0].backend: Invalid value: “”: resource or service backend is required```